pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTaskRestletResource.java

```diff
@@ -451,6 +455,48 @@ public String getSubtaskProgress(@Context HttpHeaders httpHeaders,
     }
   }
 
+  @GET
+  @Path("/tasks/subtask/state/{subTaskState}/progress")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation("Get progress of all subtasks with specified state tracked by minion worker in memory")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "Success"), @ApiResponse(code = 500, message = "Internal server error")
+  })
+  public String getSubtaskWithGivenStateProgress(@Context HttpHeaders httpHeaders,
+      @ApiParam(value = "Subtask state (UNKNOWN,IN_PROGRESS,SUCCEEDED,CANCELLED,ERROR)", required = true)
+      @PathParam("subTaskState") String subTaskState,
+      @ApiParam(value = "Minion work IDs separated by comma") @QueryParam("minionWorkerIds") @Nullable
+          String minionWorkerIds) {
+    Set<String> selectedMinionWorkers = new HashSet<>();
+    if (StringUtils.isNotEmpty(minionWorkerIds)) {
+      Collections.addAll(selectedMinionWorkers, StringUtils.split(minionWorkerIds, ','));
```

Comment:
I guess that `StringUtils.split()` would handle the trimming?

Reply:
No, it does not 😄 . Let me fix it.